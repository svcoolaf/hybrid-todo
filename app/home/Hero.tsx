import React from "react";
import styled from "styled-components";
import theme from "../common/lib/theme";

const Background = styled.div`
  padding: ${2 * theme.sizes.form}px ${theme.sizes.quadruple}px
    ${theme.sizes.quadruple}px;
  background-size: cover;
  background: linear-gradient(#7d8ca0, #506eb4);
  color: ${theme.colors.lightest};
  line-height: ${theme.sizes.h1}px;
`;

const H1 = styled.h1`
  font-weight: 200;
  font-size: ${theme.sizes.h1}px;
  padding: ${theme.sizes.half}px;
`;

const H2 = styled.h2`
  font-weight: 200;
  font-size: ${theme.sizes.h2}px;
  padding: ${theme.sizes.half}px;
`;

const Hero = () => (
  <Background>
    <H1>Today</H1>
    <H2>
      {new Intl.DateTimeFormat("en-ca", {
        weekday: "long",
        month: "long",
        day: "numeric"
      }).format(new Date())}
    </H2>
  </Background>
);

export default Hero;
