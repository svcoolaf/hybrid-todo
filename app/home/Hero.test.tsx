import React from "react";
import renderer from "react-test-renderer";
import Hero from "./Hero";

describe("Hero", () => {
  it("renders", () => {
    const tree = renderer.create(<Hero />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
