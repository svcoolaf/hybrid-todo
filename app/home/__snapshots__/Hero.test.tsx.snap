// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders 1`] = `
<div
  className="Hero__Background-hkp9da-0 dZrZsG"
>
  <h1
    className="Hero__H1-hkp9da-1 gEoyCq"
  >
    Today
  </h1>
  <h2
    className="Hero__H2-hkp9da-2 cNMwov"
  >
    Friday, August 23
  </h2>
</div>
`;
